src/Http/WebUtilities/src/MultipartPipeReader.cs

```diff
@@ -0,0 +1,432 @@
+            _boundary.ExpectLeadingCrlf = true;
+            var body = ReadBody(ref _buffer, _readResult.IsCompleted);
+            return new MultipartSection() { Headers = headersAccumulator.GetResults(), Body = new MemoryStream(body), BaseStreamOffset = 0 };
+        }
+
+
+        void SkipMetaData(ref ReadOnlySequence<byte> buffer, bool isFinalBlock)
+        {
+            var sequenceReader = new SequenceReader<byte>(buffer);
+            var consumedBytes = default(long);
+            var crlfDelimiter = new byte[] { (byte)'\r', (byte)'\n' };
+            var endOfFileDelimiter = new byte[] { (byte)'-', (byte)'-' };
+
+            if (!sequenceReader.TryReadTo(out var body, crlfDelimiter))
+            {
+                if (sequenceReader.TryReadTo(out var _, endOfFileDelimiter))
+                {
+                    buffer = buffer.Slice(sequenceReader.Position);
+                    return;
+                }
+
+                if (!isFinalBlock)
+                {
+                    // Don't buffer indefinately
+                    if ((sequenceReader.Consumed - consumedBytes) > 100)
+                    {
+                        Debug.Print("Un-expected data found on the boundary line: " + body);
+                        return;
+                    }
+                }
+            }
+            var position = sequenceReader.Position;
+            var consumed = sequenceReader.Consumed;
+            sequenceReader.Rewind(sequenceReader.Consumed);
+            if (sequenceReader.TryReadTo(out var _, endOfFileDelimiter))
+            {
+                if (sequenceReader.Consumed < consumed)
+                {
+                    buffer = buffer.Slice(sequenceReader.Position);
+                }
+            }
+            buffer = buffer.Slice(position);
+
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        internal void ParseHeaders(
+            ref ReadOnlySequence<byte> buffer,
+            ref KeyValueAccumulator accumulator,
+            bool isFinalBlock)
+        {
+            if (buffer.IsSingleSegment)
+            {
+                ParseHeadersFast(buffer.First.Span,
+                    ref accumulator,
+                    isFinalBlock,
+                    out var consumed);
+
+                buffer = buffer.Slice(consumed);
+                return;
+            }
+
+            ParseValuesSlow(ref buffer,
+                ref accumulator,
+                isFinalBlock);
+        }
+
+        // Fast parsing for single span in ReadOnlySequence
+        private void ParseHeadersFast(ReadOnlySpan<byte> span,
+            ref KeyValueAccumulator accumulator,
+            bool isFinalBlock,
+            out int consumed)
+        {
+            ReadOnlySpan<byte> key;
+            ReadOnlySpan<byte> value;
+            consumed = 0;
+            var colonDelimiter = new byte[] { (byte)':' };
+            var crlfDelimiter = new byte[] { (byte)'\r', (byte)'\n' };
```

Comment:
```suggestion
            ReadOnlySpan<byte> crlfDelimiter = new byte[] { (byte)'\r', (byte)'\n' };
```

As the `crlfDelimiter` is used in several places, move it to a static field. Avoids duplication, codegen / perf is the same.
```c#
private static ReadOnlySpan<byte> CrlfDelimiter => new byte[] { (byte)'\r', (byte)'\n' };
```